redbeams/src/main/java/com/sequenceiq/redbeams/service/stack/DBStackUpdater.java

```diff
@@ -0,0 +1,47 @@
+package com.sequenceiq.redbeams.service.stack;
+
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import javax.inject.Inject;
+
+import org.springframework.stereotype.Service;
+
+import com.sequenceiq.cloudbreak.common.exception.NotFoundException;
+import com.sequenceiq.redbeams.api.endpoint.v4.databaseserver.responses.SslCertificateType;
+import com.sequenceiq.redbeams.configuration.DatabaseServerSslCertificateConfig;
+import com.sequenceiq.redbeams.configuration.SslCertificateEntry;
+import com.sequenceiq.redbeams.domain.stack.DBStack;
+import com.sequenceiq.redbeams.domain.stack.SslConfig;
+
+@Service
+public class DBStackUpdater {
+
+    @Inject
+    private DatabaseServerSslCertificateConfig databaseServerSslCertificateConfig;
+
+    @Inject
+    private DBStackService dbStackService;
+
+    public void updateSslConfig(long id) {
+        Optional<DBStack> dbStackOpt = dbStackService.findById(id);
+        dbStackOpt.ifPresent(dbStack -> {
+            SslConfig sslConfig = dbStack.getSslConfig();
+            if (sslConfig != null && sslConfig.getSslCertificateType() == SslCertificateType.CLOUD_PROVIDER_OWNED) {
+                String cloudPlatform = dbStack.getCloudPlatform();
+                String region = dbStack.getRegion();
+                Set<SslCertificateEntry> allCertificates = databaseServerSslCertificateConfig.getCertsByCloudPlatformAndRegion(cloudPlatform, region);
+                sslConfig.setSslCertificates(allCertificates.stream().map(SslCertificateEntry::getCertPem).collect(Collectors.toSet()));
```

Comment:
This could maybe also get extracted to another method as getCertPems, in DatabaseServerSslCertificateConfig.